src/address_finder.rs

```diff
@@ -71,17 +71,22 @@ mod os_impl {
     use elf;
     use proc_maps::*;
     use failure::Error;
+    use failure::ResultExt;
     use libc::pid_t;
+    use libc;
     use std;
+    use std::fs;
+    use std::os::linux::fs::MetadataExt;
     use address_finder::AddressFinderError;
     use read_process_memory::*;
+    use std::os::unix::prelude::*;
 
     pub fn current_thread_address(
         pid: pid_t,
         version: &str,
         is_maybe_thread: Box<Fn(usize, &ProcessHandle, &Vec<MapRange>) -> bool>,
     ) -> Result<usize, Error> {
-        let proginfo = &get_program_info(pid)?;
+        let proginfo = &get_program_info_wrapper(pid)?;
```

Comment:
I'd leave this named as is, and call the other one something like `_get_program_info` or `inner` or something.